### PR TITLE
Added filter to work item list (fixes #376)

### LINKED ISSUE #376: Button or Menu to access the filter "assigned to me"

### PR DESCRIPTION
**Pull request by sanbornsen**
- Added filter UI to the list page
- Added resolver to fetch list of users
- Added resolver to fetch the auth user details
- Removed user service dependency from header
- Updated the algorithm of infinite scroll directive to support list
reload.

### CODE REVIEW

File: src/app/in-memory-data.service.ts [hunk 1]
```diff
@@ -46,6 +46,14 @@ export class InMemoryDataService implements InMemoryDbService {
         },
         'type': 'workitems'
       };
+      // return {'fields': {'system.assignee': 'someUser' + n,
```
Comment (joshuawilson) [inline src/app/in-memory-data.service.ts]: Is this commented code from the last merge?

Comment (sanbornsen) [inline src/app/in-memory-data.service.ts]: Yes !! There was no conflict so didn't look at it.

File: src/app/work-item/users.resolver.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { Resolve, ActivatedRouteSnapshot } from '@angular/router';
+import { UserService } from './../user/user.service';
```
Comment (joshuawilson) [inline src/app/work-item/users.resolver.ts]: more System.js path styles

File: src/app/work-item/work-item-list/work-item-list.component.ts [hunk 3]
```diff
@@ -9,6 +25,9 @@ import { Logger } from '../../shared/logger.service';
 import { WorkItem }                   from '../../models/work-item';
 import { WorkItemListEntryComponent } from './work-item-list-entry/work-item-list-entry.component';
 import { WorkItemService }            from '../work-item.service';
+import { UserService } from '../../user/user.service';
+import { User } from './../../models/user';
```
Comment (joshuawilson) [inline src/app/work-item/work-item-list/work-item-list.component.ts]: System.js path style

File: src/app/work-item/work-item-list/work-item-list-routing.module.ts [hunk 4]
```diff
@@ -8,13 +9,21 @@ const routes: Routes = [
   {
     path: 'work-item-list',
     component: WorkItemListComponent,
+    resolve: {
+      allusers: UsersResolve,
+      authuser: AuthUserResolve
```
Comment (sanbornsen) [inline src/app/work-item/work-item-list/work-item-list-routing.module.ts]: Yeah, I love this feature.

File: src/app/work-item/work-item-list/work-item-list.component.ts [hunk 5]
```diff
@@ -38,30 +61,96 @@ export class WorkItemListComponent implements OnInit {
...
+    let allUsers = cloneDeep(this.route.snapshot.data['allusers']);
+    loggedInUser = allUsers.find((user: User) => user.attributes.imageURL == loggedInUser.imageURL);
+    if (loggedInUser) {
+      this.filters[0].value = loggedInUser.id;
+    }
+  }
+
+  activeFilter(filterId: number) {
+    let selectedIndex = this.filters.findIndex((f: any) => {
+      return f.id == filterId;
```
Comment (joshuawilson) [inline src/app/work-item/work-item-list/work-item-list.component.ts]: is the ID here at risk of not being the same type?

File: src/app/work-item/work-item-list/work-item-list.module.ts [hunk 6]
```diff
@@ -12,6 +13,10 @@ import { WorkItemQuickAddModule } from '../work-item-quick-add/work-item-quick-a
 import { WorkItemListEntryComponent } from './work-item-list-entry/work-item-list-entry.component';
 import { WorkItemListRoutingModule } from './work-item-list-routing.module';
 
+import { UserService } from '../../user/user.service';
+import { UsersResolve, AuthUserResolve } from './../users.resolver';
```
Comment (joshuawilson) [inline src/app/work-item/work-item-list/work-item-list.module.ts]: System.js path style

**Pull request merged**